src/store/modules/ModalsModule.ts

```diff
@@ -1,17 +1,29 @@
+import R2Error from "../../model/errors/R2Error";
 import ThunderstoreMod from "../../model/ThunderstoreMod";
+import LoggerProvider, { LogSeverity } from '../../providers/ror2/logging/LoggerProvider';
 
 interface State {
     downloadModModalMod: ThunderstoreMod | null;
+    errorModalError: R2Error | null;
     isCategoryFilterModalOpen: boolean;
     isDownloadModModalOpen: boolean;
+    isErrorModalOpen: boolean;
     isGameRunningModalOpen: boolean;
 }
 
+interface ErrorWithLogging {
+    error: R2Error;
+    severity: LogSeverity;
+    logMessage?: string;
+}
+
 export default {
     state: (): State => ({
         downloadModModalMod: null,
+        errorModalError: null,
         isCategoryFilterModalOpen: false,
         isDownloadModModalOpen: false,
+        isErrorModalOpen: false,
         isGameRunningModalOpen: false,
     }),
```

Comment:
what's the benefit in tracking the error state & modal open state separately? as far as I can tell those two fields are supposed to always be in sync anyway, could we not just check if the error exists? would eliminate any possibility of the two desyncing & potentially leading to weird UI bugs down the line.

Reply:
Just to stay consistent with `downloadModModalMod` and `disableModModalMod` (not pictured). But I agree, it's unnecessary.